cmd/nfs-subdir-external-provisioner/provisioner.go

```diff
@@ -85,31 +83,27 @@ func (p *nfsProvisioner) Provision(ctx context.Context, options controller.Provi
 	}
 	glog.V(4).Infof("nfs provisioner: VolumeOptions %v", options)
 
-	pvcNamespace := options.PVC.Namespace
-	pvcName := options.PVC.Name
-
-	pvName := strings.Join([]string{pvcNamespace, pvcName, options.PVName}, "-")
-
 	metadata := &pvcMetadata{
 		data: map[string]string{
-			"name":      pvcName,
-			"namespace": pvcNamespace,
+			"name":      options.PVC.Name,
+			"namespace": options.PVC.Namespace,
+			"pvname":    options.PVName,
 		},
 		labels:      options.PVC.Labels,
 		annotations: options.PVC.Annotations,
 	}
 
-	fullPath := filepath.Join(mountPath, pvName)
-	path := filepath.Join(p.path, pvName)
-
 	pathPattern, exists := options.StorageClass.Parameters["pathPattern"]
-	if exists {
-		customPath := metadata.stringParser(pathPattern)
-		if customPath != "" {
-			path = filepath.Join(p.path, customPath)
-			fullPath = filepath.Join(mountPath, customPath)
-		}
+	if !exists {
```

Comment:
to prevent my previous question about empty pathPattern, you could do `if !exists || pathPattern == "" {`

Reply:
Done. This will make it more error proof.